karapace/schema_registry_apis.py

```diff
@@ -756,6 +757,10 @@ async def subjects_schema_post(
             validated_typed_schema = ValidatedTypedSchema.parse(schema["schema"].schema_type, schema["schema"].schema_str)
             if schema_type is SchemaType.JSONSCHEMA:
                 schema_valid = validated_typed_schema.to_dict() == new_schema.to_dict()
+            elif schema_type is SchemaType.PROTOBUF:
+                result = CompareResult()
+                validated_typed_schema.schema.compare(new_schema.schema, result)
+                schema_valid = result.is_compatible()
```

Comment:
`compare` + `is_compatible` are wrong here. Those compare against backward compatibility of schemas, not equality. E.g. addition of new (non-required) field is compatible change, but not equal schema anymore. This is wrong. But you correctly catched here, the original code seems to be wrong also...